packages/react/src/products/hooks/__tests__/useProductGrouping.test.tsx

```diff
@@ -0,0 +1,184 @@
+  fetchProductGrouping: jest.fn(() => () => Promise.resolve()),
+}));
+
+describe('useProductGrouping', () => {
+  beforeEach(jest.clearAllMocks);
+
+  it('should return data correctly with initial state', () => {
+    const { result } = renderHook(() => useProductGrouping(mockProductId), {
+      wrapper: withStore(mockProductsState),
+    });
+
+    expect(result.current).toStrictEqual({
+      error: null,
+      isFetched: true,
+      isLoading: false,
+      data: mockProductGroupingAdapted,
+      actions: {
+        refetch: expect.any(Function),
+      },
+    });
+  });
+
+  it('should return error state', () => {
+    const errorMockProductsState = {
+      entities: {},
+      products: {
+        grouping: {
+          error: {
+            [mockProductId]: 'Error - Not loaded.',
+          },
+          isLoading: {
+            [mockProductId]: false,
+          },
+        },
+      },
+    };
+
+    const { result } = renderHook(() => useProductGrouping(mockProductId), {
+      wrapper: withStore(errorMockProductsState),
+    });
+
+    expect(result.current).toStrictEqual({
+      error: 'Error - Not loaded.',
+      isFetched: true,
+      isLoading: false,
+      data: undefined,
+      actions: {
+        refetch: expect.any(Function),
+      },
+    });
+  });
+
+  it('should return loading state', () => {
+    const loadingMockProductsState = {
+      entities: {},
+      products: {
+        grouping: {
+          error: {},
+          isLoading: {
+            [mockProductId]: true,
+          },
+        },
+      },
+    };
+
+    const { result } = renderHook(() => useProductGrouping(mockProductId), {
+      wrapper: withStore(loadingMockProductsState),
+    });
+
+    expect(result.current).toStrictEqual({
+      error: undefined,
+      isFetched: false,
+      isLoading: true,
+      data: undefined,
+      actions: {
+        refetch: expect.any(Function),
+      },
+    });
+  });
+
+  describe('options', () => {
+    it('should fetch data if `enableAutoFetch` option is true', () => {
+      const initialMockProductsState = {
+        entities: {},
+        products: {
+          grouping: {
+            error: {},
+            isLoading: {},
+          },
+        },
+      };
+
+      const options = {
+        fetchQuery: {},
+        fetchConfig: { headers: { 'X-Test-Header': 'test' } },
+      };
+
+      renderHook(() => useProductGrouping(mockProductId, options), {
+        wrapper: withStore(initialMockProductsState),
+      });
+
+      expect(fetchProductGrouping).toHaveBeenCalledWith(
+        mockProductId,
+        options.fetchQuery,
+        options.fetchConfig,
+      );
+    });
+
+    it('should not fetch data if `enableAutoFetch` option is false', () => {
+      renderHook(
+        () =>
+          useProductGrouping(mockProductId, {
+            enableAutoFetch: false,
+          }),
+        {
+          wrapper: withStore(mockProductsState),
```

Comment:
This state corresponds to an already loaded state, which means that even if `enableAutoFetch` was true, it would not make a fetch request. To be correct, this test should use a state where the product grouping is not loaded yet to make sure that the fetch request was not made because `enableAutoFetch` was false and not that data was already loaded.